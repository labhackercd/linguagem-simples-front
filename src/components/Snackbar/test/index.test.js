
import React from 'react';
import {shallow} from "enzyme/build";

import { mount} from 'enzyme';
import MockAdapter from "axios-mock-adapter"
import CustomizedSnackbars from '../'

it("Should render the SavedContentComponent", () => {
    const component = shallow(<CustomizedSnackbars open={true} message={"Mensagem Teste"} type={"error"}></CustomizedSnackbars>);

    expect(component.exists()).toEqual(true);
    expect(component).toMatchSnapshot();
});


describe('Testing render of different types', () => {

    test("Test if sucess alert is rendered", async (done) => {
          
        const  wrapper = mount(<CustomizedSnackbars open={true} message={"Mensagem Teste"} type={"success"}></CustomizedSnackbars>);
        expect(wrapper.exists()).toEqual(true);

        done();
    });

    test("Test if error alert is rendered", async (done) => {
      
        const  wrapper = mount(<CustomizedSnackbars open={true} message={"Mensagem Teste"} type={"error"}></CustomizedSnackbars>);
        expect(wrapper.exists()).toEqual(true);

        done();
    });

    test("Test handle close", async (done) => {
      
        const  wrapper = mount(<CustomizedSnackbars open={true} message={"Mensagem Teste"} type={"error"}></CustomizedSnackbars>);
        const button = wrapper.find("button").at(0);
        button.simulate('click');
        done();
    });
});