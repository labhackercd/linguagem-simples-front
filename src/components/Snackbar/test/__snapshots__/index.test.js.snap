// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the SavedContentComponent 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Snackbar))
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "bottom",
      }
    }
    autoHideDuration={6000}
    bodystyle={
      Object {
        "height": "auto",
        "maxWidth": "100%",
      }
    }
    onClose={[Function]}
    open={true}
  >
    <Alert
      onClose={[Function]}
      severity="error"
    >
      Mensagem Teste
    </Alert>
  </WithStyles(ForwardRef(Snackbar))>
</div>
`;
